### PR TITLE
Deprecate  `-gui` and continuing the flow with an existing tag

### PR DESCRIPTION
**Pull request by kareefardi**
`-gui` is superseded by `gui.py`. `or_gui` should still work for interactive scripts. 
Dropping support for continuing existing runs. Users need to use `-overwrite` or provide a different tag.

### CODE REVIEW

**Comment by mguthaus** [comment]
Dropping -overwrite requires that a user manually delete a run directory every time they want to re-run with a tag. otherwise, the run will continue starting with the last step of the previous run. I believe this wll result in quite a bit of user confusion.

Suggestion: 
Either keep -overwrite or fail if a run directory of a given tag already exists.

**Comment by mguthaus** [comment]
Defaulting to overwrite without warning could be catastrophic, and is not recommended.

**Comment by kareefardi** [comment]
I wanted to drop support for rerunning existing tag, but I guess I can do that while requiring `-overwrite` to re-use a tag.

File: flow.tcl [hunk 1]
```diff
@@ -165,13 +165,14 @@ proc run_non_interactive_mode {args} {
         exit -1
     }
 
-    prep {*}$args
-    # signal trap SIGINT save_state;
-
     if { [info exists flags_map(-gui)] } {
-        or_gui
+        puts_warn "Flag -gui is now deprecated. Refer to https://openlane.readthedocs.io/en/latest/reference/gui.html to view files graphically."
         return
```
Comment (donn) [inline flow.tcl:170]: Exit should be non-zero.

**Review by donn — CHANGES_REQUESTED** [review]

**Review by donn — APPROVED** [review]

**Pull request merged**